xesmf/frontend.py

```diff
@@ -509,3 +475,13 @@ def regrid_dataset(self, ds_in, keep_attrs=False):
             ds_out = ds_out.squeeze(dim='dummy')
 
         return ds_out
+
+    def to_netcdf(self, filename=None):
+        '''Save weights to disk as a netCDF file.'''
+        if filename is None:
+            filename = self._get_default_filename()
+        w = self.weights
+        ds = xr.Dataset({"S": w.data, "col": w.col + 1, "row": w.row + 1})
```

Comment:
Fortran vs C indexing?

Reply:
As far as I can tell, this reproduces how weights are currently stored by ESMF. 
@bekozi Is anything missing ?

Reply:
Why not use ESMF to write the weights? If this is necessary, write a test using an ESMF-generated weight file and an xESMF-generated weight file and assert the contents are equivalent.

Reply:
To clarify, the idea is to pass the weights from ESMPy to xESMF in memory, then as needed by the user, write them to disk. This constrats with the current behavior of writing weights to disk then have xESMF read them. 

@bekozi Given a sparse COO weights matrix, can I pass it to ESMPy and ask it to store it to a given netCDF file path ?

If not, I'll write a test as you suggest.

Reply:
I see. In that case, it doesn't make sense to use esmf to write the weights. If you want the weight files exactly the same, the factor count dimension should be named `n_s`. This is the bare bones weight file structure. There is an option to output a bunch of auxiliary variables in the weight file but that is turned off by default.

Reply:
Thanks, dimension added. Datasets are now identical according to `xr.testing.assert_identical`.